x/auction/client/cli/query.go

```diff
@@ -68,30 +78,97 @@ func QueryGetAuctionCmd(queryRoute string, cdc *codec.Codec) *cobra.Command {
-		Short: "get a list of active auctions",
-		Args:  cobra.NoArgs,
+		Short: "query auctions with optional filters",
+		Long: strings.TrimSpace(`Query for all paginated auctions that match optional filters:
+Example:
+$ kvcli q auction auctions --type=(Collateral|Surplus|Debt)
+$ kvcli q auction auctions --denom=bnb
+$ kvcli q auction auctions --phase=(Forward|Reverse)
+$ kvcli q auction auctions --page=2 --limit=100
+`,
+		),
 		RunE: func(cmd *cobra.Command, args []string) error {
+			strType := viper.GetString(flagType)
+			strDenom := viper.GetString(flagDenom)
+			strPhase := viper.GetString(flagPhase)
+			page := viper.GetInt(flags.FlagPage)
+			limit := viper.GetInt(flags.FlagLimit)
+
+			var auctionType string
+			var auctionDenom string
+			var auctionPhase string
+
+			params := types.NewQueryAllAuctionParams(page, limit, auctionType, auctionDenom, auctionPhase)
+
+			if len(strType) != 0 {
+				auctionType = strings.ToLower(strings.TrimSpace(strType))
+				if auctionType != types.CollateralAuctionType &&
+					auctionType != types.SurplusAuctionType &&
+					auctionType != types.DebtAuctionType {
+					return fmt.Errorf("invalid auction type %s", strType)
+				}
+				params.Type = auctionType
+			}
+
+			if len(strDenom) != 0 {
+				auctionDenom := strings.TrimSpace(strDenom)
+				err := sdk.ValidateDenom(auctionDenom)
+				if err != nil {
+					return err
+				}
+				params.Denom = auctionDenom
+			}
+
+			if len(strPhase) != 0 {
+				auctionPhase := strings.ToLower(strings.TrimSpace(strPhase))
+				if auctionType != types.CollateralAuctionType && len(auctionType) > 0 {
+					return fmt.Errorf("cannot apply phase flag to non-collateral auction type")
+				}
+				if auctionPhase != types.ForwardAuctionPhase && auctionPhase != types.ReverseAuctionPhase {
+					return fmt.Errorf("invalid auction phase %s", strPhase)
+				}
+				params.Phase = auctionPhase
+			}
+
+			bz, err := cdc.MarshalJSON(params)
+			if err != nil {
+				return err
+			}
+
 			cliCtx := context.NewCLIContext().WithCodec(cdc)
 
 			// Query
-			res, _, err := cliCtx.QueryWithData(fmt.Sprintf("custom/%s/%s", queryRoute, types.QueryGetAuctions), nil)
+			res, height, err := cliCtx.QueryWithData(fmt.Sprintf("custom/%s/%s", queryRoute, types.QueryGetAuctions), bz)
 			if err != nil {
 				return err
 			}
 
 			// Decode and print results
-			var auctions types.Auctions
-			cdc.MustUnmarshalJSON(res, &auctions)
+			var matchingAuctions types.Auctions
+			cdc.MustUnmarshalJSON(res, &matchingAuctions)
+
+			if len(matchingAuctions) == 0 {
+				return fmt.Errorf("No matching auctions found")
+			}
 
 			auctionsWithPhase := []types.AuctionWithPhase{} // using empty slice so json returns [] instead of null when there's no auctions
-			for _, a := range auctions {
+			for _, a := range matchingAuctions {
 				auctionsWithPhase = append(auctionsWithPhase, types.NewAuctionWithPhase(a))
 			}
-			return cliCtx.PrintOutput(auctionsWithPhase)
+			cliCtx = cliCtx.WithHeight(height)
```

Comment:
we should add this too for the other auction queries

Reply:
https://github.com/Kava-Labs/kava/pull/522/commits/94c1ebd775c02c8f01c1ad32745b29d4ba58d4f5